src/gpu.rs

```diff
@@ -250,3 +262,53 @@ pub fn msaa_samples_limited(physical_device: &PhysicalDevice, target_msaa_sample
     let msaa_limit = std::cmp::min(color_limit, depth_limit);
     std::cmp::min(msaa_limit, target_msaa_samples)
 }
+
+#[cfg(all(target_os = "macos", not(test)))]
+pub fn check_moltenvk(vulkan_builder: VulkanInstanceBuilder, settings: Option<moltenvk_deps::Install>) -> VulkanInstanceBuilder {
+    let settings = match settings {
+        Some(s) => s,
+        None => Default::default(),
+    };
+    let path = match moltenvk_deps::check_or_install(settings) {
+        Err(moltenvk_deps::Error::ResetEnvVars(p)) => Some(p),
+        Err(moltenvk_deps::Error::NonDefaultDir) => None,
+        Err(moltenvk_deps::Error::ChoseNotToInstall) => panic!("Moltenvk is required for Nannou on MacOS"),
+        Err(e) => panic!("Moltenvk installation failed {:?}", e),
+        Ok(p) => Some(p),
+    };
+    let loader = path.map(|p| {
+        unsafe { DynamicLibraryLoader::new(p) }
+    });
+    match loader {
+        Some(Ok(l)) => {
+            let loader: FunctionPointers<Box<(dyn Loader + Send + Sync + 'static)>> = FunctionPointers::new(Box::new(l));
+            let required_extensions = required_extensions_with_loader(&loader);
+            vulkan_builder.extensions(required_extensions)
+            .add_loader(loader)
+        },
+        _ => vulkan_builder,
+    }
+}
+
+
+pub fn required_extensions_with_loader<L>(ptrs: &FunctionPointers<L>)
+    -> InstanceExtensions 
+    where L: Loader
+{
+    let ideal = InstanceExtensions {
+        khr_surface: true,
+        khr_xlib_surface: true,
+        khr_xcb_surface: true,
+        khr_wayland_surface: true,
+        khr_android_surface: true,
+        khr_win32_surface: true,
+        mvk_ios_surface: true,
+        mvk_macos_surface: true,
+        ..InstanceExtensions::none()
+    };
```

Comment:
This looks like the same extensions as required by `vulkano-win` [here](https://github.com/vulkano-rs/vulkano/blob/master/vulkano-win/src/lib.rs#L30) - should we be using `vulkano_win::required_extensions()` instead? Where did you get this list from?

I'm not 100% sure I understand this function - it says `required_extensions_with_loader` but then it returns the intersection of ideal and supported extensions?

Reply:
Ahhh I see, this is a copy of `vulkano_win::required_extensions` but your checking what's supported by core using the custom loader?

Reply:
It is indeed the same function except that you can give it a loader. The one in `vulkano_win` uses `auto_loader` which will not work when the env variables are not sourced. So in order to allow installation to work without requiring the user to source `.bash_profile` I need to provide a loader with a specific path to the lib. This loader also needs to be user to get the `required_extensions` unfortunately `vulkano_win` does not provide this ability.

Another odd fact is that everything in that function is actually just in `vulkano` so I think it's a little odd that it is even in `vulkano-win`